src/Parser/Core/Modules/SpellHistory.js

```diff
@@ -42,6 +47,22 @@ class SpellHistory extends Analyzer {
     }
   }
 
+  _addTimeWaitingOnGCD(event) {
+    if (!this.lastGlobalCooldown) {
+      return event;
+    }
+    const resetBufferMS = RESET_BUFFER_PERCENT * this.lastGlobalCooldown.duration;
+    const earlyByMS = event.start + event.expectedDuration - event.timestamp;
+    // check if the ability was reset early. If the reset was less than a percentage of the GCD don't consider it as a reset.
+    // check if the reset happened as a result of the previous cast
+    if (event.trigger === 'endcooldown' && resetBufferMS < earlyByMS 
```

Comment:
This if-statement has a bunch of conditionals making it harder to read that could be separated into multiple if-statements with early returns instead.